context-free.md

```diff
@@ -1,22 +1,44 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+ContextFree can be used by people who want to create digital images, 
+especially those that involve repeating structures, through some simple
+programming.
 
 ## What is easy to do in this language? Why is it easy?
+It is quite easy to create images that involve many repeating shapes because 
+you can create multiples of many shapes and manipulate them using recursion.
 
-
-## What is hard to do in this language? Why is it hard?
-
+## What is hard to do in this language? Why is it hard? 
+It is more difficult to create organic shapes and lines in this language 
+because you are essentially limited to three basic shapes: squares, circles, 
+and triangles. It is much different from physically drawing something because 
+you need to program every part of the image. So while it is very easy to make
+repetitive patterns it is hard to create something that looks more free form 
+and natural.
 
 ## How did you learn how to program in this language?
-
+I learned through the two lessons and the examples that were included in the 
+ContextFree program, through the documentation on the ContextFree website, 
+and as always, by simply playing around with the program and making many 
+mistakes!
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
-
+The user is basically creating images by writing code that corresponds to the 
+output of shapes in different formations. Blocks of code, called “rules”, 
+each describe some element of the image. These rules can be called within 
+other rules, and you can even call a rule within itself, allowing for 
+recursive images (the recursion ends when the shape becomes too small to see).
+The main rule takes on the name of the “startshape”, and all other rules must 
+be called within that main rule in order to be used.
 
 ## What do you think is interesting about the ContextFree program you wrote?
```

Comment:
Peacocks are beautiful, and you did a great job of depicting one using ContextFree! I thought it was great that you depicted a recognizable animal rather than just a cool pattern.